1. Importe numpy como 'np' e imprima o número da versão instalada no ambiente (local ou colab).

In [5]:
import numpy as np
np.version

<module 'numpy.version' from 'c:\\Users\\Mirella\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\numpy\\version.py'>

2. Crie um *array* 1D com números de 0 a 9 e imprima o resultado.

In [6]:
a = np.array(range(10))
print(a)
a

[0 1 2 3 4 5 6 7 8 9]


array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

Crie uma matriz booleana de 3x3 e preencha os seus valores com 'True'. Imprima o resultado.
Dica: Verifique o uso da função full na documentação do numpy.

In [7]:
matriz_bool = np.full ((3, 3), True)
print(matriz_bool)
matriz_bool

[[ True  True  True]
 [ True  True  True]
 [ True  True  True]]


array([[ True,  True,  True],
       [ True,  True,  True],
       [ True,  True,  True]])

4. Crie um *array* 1D como números inteiros (50 elementos) de 0 a 1000 e **extraia** apenas os números pares. Imprima o resultado.

In [8]:
num_int = np.random.randint(0, 1000, 50)
print(num_int)
pares = num_int[num_int%2==0]
print(pares)

[ 18 670  73 447 580 835 897 855 270 763 860 453 376 901 455 914 523 547
 779 317 143 767 874  58 727 362 372 325 436 582 421 993   0 357 441  62
 862 106  18  73  62 604 222 877 267 341 509 376 783  37]
[ 18 670 580 270 860 376 914 874  58 362 372 436 582   0  62 862 106  18
  62 604 222 376]


5. Usando o *array* criando no exercício anterior, **substitua** todos os números ímpares maiores que 50 por -1 no *array* original. 

In [9]:
num50 = num_int[num_int > 50]
num50[num50 %2 != 0] = -1
num50

array([670,  -1,  -1, 580,  -1,  -1,  -1, 270,  -1, 860,  -1, 376,  -1,
        -1, 914,  -1,  -1,  -1,  -1,  -1,  -1, 874,  58,  -1, 362, 372,
        -1, 436, 582,  -1,  -1,  -1,  -1,  62, 862, 106,  -1,  62, 604,
       222,  -1,  -1,  -1,  -1, 376,  -1])

6. Usando o *array* criando no exercício anterior, **substitua** todos os números ímpares com o valor -1 **sem** alterar o *array* original.

In [16]:
num_int2 = num_int.copy()
num_int2[num_int%2!=0] = -1
num_int2

array([ 18, 670,  -1,  -1, 580,  -1,  -1,  -1, 270,  -1, 860,  -1, 376,
        -1,  -1, 914,  -1,  -1,  -1,  -1,  -1,  -1, 874,  58,  -1, 362,
       372,  -1, 436, 582,  -1,  -1,   0,  -1,  -1,  62, 862, 106,  18,
        -1,  62, 604, 222,  -1,  -1,  -1,  -1, 376,  -1,  -1])

7. Crie um *array* 1D com 100 valores de ponto flutuante (**float**), converta-o em uma matriz 2D com 10 linhas e 10 colunas.

In [18]:
vetor_float = np.round((np.random.rand(100)*100), 2)
vetor_float.reshape((10,10))

array([[53.63, 42.67, 42.83, 74.03, 26.93, 70.26, 51.12, 89.29, 17.17,
        69.81],
       [94.68, 37.13, 76.79, 56.46,  0.82, 47.68, 98.  , 80.96, 54.84,
        63.71],
       [57.27, 99.69, 69.18, 81.4 , 90.01, 90.65, 37.49,  8.43, 73.89,
        83.48],
       [ 3.03, 10.33, 27.05, 56.99, 11.17, 85.22, 48.91, 14.12, 92.8 ,
        50.45],
       [ 9.57, 70.03, 91.75, 92.4 , 57.4 , 57.38, 88.15, 68.99, 72.57,
        92.46],
       [89.55, 13.25, 48.39, 61.09, 68.55, 52.02, 63.44, 33.36, 29.25,
        86.53],
       [69.91, 37.16, 53.98, 24.92, 50.14,  4.51, 57.98, 43.27, 23.87,
         5.04],
       [66.43, 72.61, 86.63, 60.24, 18.08, 80.93, 38.04, 34.97, 23.32,
        97.53],
       [98.08, 57.97, 54.64, 28.87, 76.46, 54.06, 35.69, 54.47, 36.03,
        65.17],
       [66.15, 17.91, 78.74, 95.87, 89.86,  9.59, 99.78, 23.76, 94.15,
        84.17]])

Crie duas matrizes 2D de inteiros aleatórios (de 0 a 100) com dimensões 10x10. Empilhe-as horizontalmente.
Dica: verique as funções hstack e concatenate na documentação do numpy

In [19]:
matriz1 = np.random.randint(0, 100, (10,10))
matriz2 = np.random.randint(0, 100, (10,10))
vertical = np.vstack((matriz1, matriz2))
horizontal = np.hstack((matriz1, matriz2))
print(horizontal)


[[54 98 48  8  6 75 16 36  3 78 53 38 25 57 15  5 81 93 43 64]
 [36 40 76 93 48 32  1 82 45 17 83 73 72 41 73 83  2 52 79 26]
 [ 2 47 26 75 43 27 72 83 24 29 45  3 67 36 31 46 17 68  0  2]
 [ 4 17 55 71 98 15 85 21 36  9 34 71 43 93 58 68 42 63 76 63]
 [18 45 29 67 75 90 98 51  9  2 95 97 23 95 83 84 87 27 81 75]
 [74 86 29 30 16 43 45 27 19 17 38 54 35 90 73 98 62 58 99 83]
 [67 26 16 85 16 17 28 70 75  6 32 85 44 39 93 53  2 10 86 38]
 [78 69 68  7 31 16 98 31 88 30 48  4 44 83 69 61 96 43 26 70]
 [70 21 61 97 36 83 20 88 71 71 26 32 52 70  4 94 10 49 59 32]
 [34 92 37 67 18 53 97  2 36 51 75 87 95 39 70 57 73 33  3 67]]


Crie duas matrizes 2D de inteiros aleatórios (de 0 a 100) com dimensões 10x10. Empilhe-as verticalmente.
Dica: verique as funções vstack e concatenate na documentação do numpy

In [13]:
vertical

array([[41, 90, 74, 92, 39,  8, 70, 36, 19, 39],
       [50, 81, 59, 34, 86, 41,  3, 10, 38, 24],
       [98, 31, 75,  5,  9, 95,  4, 52, 88,  4],
       [55, 73, 25, 94, 43, 60, 39, 75, 91, 76],
       [23, 69, 26, 34, 49, 50,  2,  5, 60, 16],
       [28, 34, 42, 64, 57, 96, 46, 42, 50, 66],
       [41, 22, 29, 57, 23, 59, 85, 42, 50, 27],
       [56, 99, 96, 93, 42, 96, 87, 91, 66, 63],
       [13, 20, 85, 60, 35, 21, 83, 97, 96, 39],
       [58, 49, 91, 31, 77, 33, 98, 27, 59, 56],
       [67,  2, 33, 42, 58, 69, 79, 96, 18, 96],
       [93, 64, 26, 86, 34, 25, 39, 21, 76,  4],
       [72, 50, 95, 60, 24, 47,  1, 35,  5, 46],
       [40, 94, 42, 28, 12, 52, 27, 62, 49, 49],
       [22, 74, 16, 71, 69, 89, 59, 20, 99, 47],
       [58, 12, 37, 47, 31, 66, 13,  4, 74, 67],
       [51, 60,  7,  3, 61, 51,  8, 46,  0, 52],
       [90, 78,  2, 31, 79, 83, 38, 98, 12, 69],
       [75, 90, 11, 27, 41, 10, 62, 62, 60, 29],
       [30, 47,  2, 47, 65, 88, 58, 13, 88, 55]])

Para ler arquivos em formato txt que estão em formato de "tabelas", podemos usar a função genfromtxt() do numpy. Esta função recebe como argumentos: caminho do arquivo txt (path local ou uma url); delimiter para separar cada valor seguindo algum critério; dtype para converter os valores recebidos em algum tipo de dado conhecido (exemplo: float) e; usecols em que é definido o cabeçalho das colunas lidas a partir do arquivo.

Vejamos um exemplo de uso da função genfromtxt sobre a base de dados Iris.

In [14]:
import numpy as np
# Caminho da base de dados é uma url disponível na internet (visite o link abaixo
# pelo navegador para título de curiosidade :))
url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data'
# Lê os dados a partir do caminho passado e armazena em uma matriz 2D chamada iris_2d
iris_2d = np.genfromtxt(url, delimiter=',', dtype='float', usecols=[0,1,2,3])

In [15]:
# matriz 2D criada com os dados da base Iris
iris_2d

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3

11. A partir dos dados lidos na matriz iris_2d, filtre as linhas que possuem (coluna 3) > 1.5 e (coluna 1) < 5.0.

12. A partir dos dados lidos na matriz iris_2d, calcule a **média** e **desvio padrão** de cada coluna presente nos dados. Imprima o resultado.

13. Crie um repositório no Github e versione todos os códigos que desenvolveram nesta lista de exercícios.